sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormat.scala

```diff
@@ -206,6 +188,23 @@ object BinaryFileFormat {
       case _ => (_ => true)
     }
   }
+
+  private[binaryfile] def genPrunedRow(
+      path: String,
+      status: FileStatus,
+      readContent: => Array[Byte],
+      requiredFieldNames: Array[String]): InternalRow = {
+
+    val values = requiredFieldNames.map {
+      case PATH => UTF8String.fromString(path)
+      case LENGTH => status.getLen
+      case MODIFICATION_TIME => DateTimeUtils.fromMillis(status.getModificationTime)
+      case CONTENT => readContent
```

Comment:
I don't see a strong reason to prune other columns that are inexpensive. Code is much simpler if we only prune `content`.

Reply:
But I think current code is simpler.
The previous code contains some code which is hard to read:
```
          val fullOutput = dataSchema.map { f =>
            AttributeReference(f.name, f.dataType, f.nullable, f.metadata)()
          }
          val requiredOutput = fullOutput.filter { a =>
            requiredSchema.fieldNames.contains(a.name)
          }
          val requiredColumns = GenerateUnsafeProjection.generate(requiredOutput, fullOutput)
          ...
          Iterator(requiredColumns(internalRow))
```